src/openloco/window.cpp

```diff
@@ -348,7 +348,7 @@ namespace openloco::ui
         int32_t base_height = map::tile_element_height(*map_x, *map_y);
         int16_t dest_x, dest_y;
         viewport* v = this->viewports[0];
-        centre_2d_coordinates(*map_x, *map_y, base_height, &dest_x, &dest_y, v);
+        v->centre_2d_coordinates(*map_x, *map_y, base_height, &dest_x, &dest_y);
 
         // Rebase mouse position onto centre of window, and compensate for zoom level.
         int16_t rebased_x = ((this->width >> 1) - mouse_x) * (1 << v->zoom),
```

Comment:
Might be also worth to add a utility function that scales values based on zoom, this kind of code will probably pop up else where.